test/jdk/javax/swing/JInternalFrame/InternalFrameBorderTest.java

```diff
@@ -0,0 +1,248 @@
+                checkBorderMidPoints("TOP");
+                checkBorderMidPoints("RIGHT");
+                checkBorderMidPoints("BOTTOM");
+                checkBorderMidPoints("LEFT");
+            });
+
+            // Check Corners
+            SwingUtilities.invokeAndWait(() -> {
+                checkCorners("TOP_LEFT");
+                checkCorners("TOP_RIGHT");
+                checkCorners("BOTTOM_RIGHT");
+                checkCorners("BOTTOM_LEFT");
+            });
+            if (!errorLog.isEmpty()) {
+                throw new RuntimeException("Following error(s) occurred: \n"
+                        + errorLog);
+            }
+        } finally {
+            if (jFrame != null) {
+                jFrame.dispose();
+            }
+            robot.delay(500);
+        }
+    }
+
+    private static void checkBorderMidPoints(String borderDirection) {
+        int x = 0, y = 0, start = 0, stop = 0;
+        switch (borderDirection) {
+            case "TOP" -> {
+                x = iFrameLoc.x + MIDPOINT;
+                y = iFrameLoc.y + BORDER_THICKNESS;
+                start = iFrameLoc.y;
+                stop = iFrameLoc.y + BORDER_THICKNESS;
+            }
+            case "RIGHT" -> {
+                x = iFrameMaxX - BORDER_THICKNESS;
+                y = iFrameLoc.y + MIDPOINT;
+                start = iFrameMaxX - BORDER_THICKNESS;
+                stop = iFrameMaxX;
+            }
+            case "BOTTOM" -> {
+                x = iFrameLoc.x + MIDPOINT;
+                y = iFrameMaxY - BORDER_THICKNESS;
+                start = iFrameMaxY;
+                stop = iFrameMaxY - BORDER_THICKNESS;
+            }
+            case "LEFT" -> {
+                x = iFrameLoc.x;
+                y = iFrameLoc.y + MIDPOINT;
+                start = iFrameLoc.x;
+                stop = iFrameLoc.x + BORDER_THICKNESS;
+            }
+            default -> throw new IllegalStateException("Unexpected value: "
+                    + borderDirection);
+        }
+
+        boolean isVertical =  borderDirection.equals("RIGHT")
+                || borderDirection.equals("LEFT");
+        boolean isHorizontal =  borderDirection.equals("TOP")
+                || borderDirection.equals("BOTTOM");
+
+        robot.mouseMove(x, y);
+        for (int i = start; i < stop; i++) {
+            if (Color.RED.equals(robot.getPixelColor(
+                    isVertical ? i : (iFrameLoc.x + MIDPOINT),
+                    isHorizontal ? i : (iFrameLoc.y + MIDPOINT)))) {
+                        saveScreenCapture(borderDirection + "_" + uiScale + ".png");
+                        errorLog.append("uiScale: "+ uiScale +
+                                " Red background color" + " detected at "
+                                + borderDirection + " border\n");
+            }
+        }
+        robot.delay(300);
+    }
+
+    private static void checkCorners(String cornerLocation) {
+        int x, y = 0;
```

Comment:
```suggestion
        int x, y;
```
Initialising `y` _only_ is redundant.